samples/bluetooth/iso_time_sync/src/main.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/console/console.h>
+#include "iso_time_sync.h"
+
+typedef enum {
+	CIS_CENTRAL = 'c',
+	CIS_PERIPHERAL = 'p',
+	BIS_TRANSMIT = 'b',
+	BIS_RECEIVER = 'r',
+} role_t;
+
+typedef enum {
+	DIR_TX = 't',
+	DIR_RX = 'r'
+} direction_t;
+
+static role_t role_select(void)
+{
+	role_t role;
+
+	while (true) {
+		printk("Choose role - cis_central (c) / cis_peripheral (p) / bis_transmit (b) / "
+		       "bis_receiver (r) : ");
+		role = console_getchar();
+		printk("\n");
+
+		switch (role) {
+		case CIS_CENTRAL:
+		case CIS_PERIPHERAL:
+		case BIS_TRANSMIT:
+		case BIS_RECEIVER:
+			return role;
+		default:
+			printk("Invalid role selected\n");
+		}
+	}
+}
+
+static direction_t direction_select(void)
+{
+	direction_t direction;
+	while (true) {
+		printk("Choose direction - TX (t) / RX (r) : ");
+		direction = console_getchar();
+		printk("\n");
+
+		switch (direction) {
+		case DIR_TX:
+		case DIR_RX:
+			return direction;
+		default:
+			printk("Invalid direction selected\n");
+			break;
+		}
+	}
+}
+
+static uint8_t bis_index_select(void)
+{
+	uint8_t index;
+
+	while (true) {
+		printk("Choose bis index [1, %d] : ", 10);
+		index = console_getchar() - '0';
+		printk("\n");
+
+		if (index > 0 && index < 10) {
+			return index;
+		} else {
+			printk("Unsupported index selected\n");
+		}
+	}
+}
+
+int main(void)
+{
+	int err;
+
+	printk("ISO Time Sync Demo\n");
+
+	err = timed_led_toggle_init();
+	if (err != 0) {
+		printk("Error failed to init LED device for toggling\n");
+		return err;
+	}
+
+	/* Initialize the Bluetooth Subsystem */
+	err = bt_enable(NULL);
+	if (err) {
+		printk("Bluetooth init failed (err %d)\n", err);
+		return 0;
+	}
+
+	console_init();
+
+	role_t role = role_select();
+	if (role == BIS_TRANSMIT) {
+		printk("Starting BIS transmitter\n");
+		bis_transmitter_start();
+	} else if (role == BIS_RECEIVER) {
+		uint8_t index = bis_index_select();
+
+		printk("Starting BIS receiver, BIS index %d\n", index);
+		bis_receiver_start(index);
+	} else {
+		direction_t direction = direction_select();
+		if (role == CIS_CENTRAL) {
+			printk("Starting CIS central, dir: %s\n",
+			       direction == DIR_TX ? "tx" : "rx");
```

Comment:
For the purposes of demonstrating time_sync, what is the point of RXing on the central?

Reply:
We will be able to demonstrate that we can synchronize the TXing and RXing side